world/combat/combat_functions.py

```diff
@@ -1,10 +1,51 @@
+import copy
+
 from world.scenes.models import Scene, LogEntry
 from django.utils.html import escape
 import random
 import math
 from world.utilities.utilities import logger
 from world.combat.attacks import Attack, ActionResult
 from world.combat.effects import BUFFS, DEBUFFS, DEBUFFS_STANDARD, DEBUFFS_HEXES, DEBUFFS_TRANSFORMATION, AimOrFeint
+from world.combat.normals import NORMALS
+from world.arts.models import Arts
+
+
+class ArtBaseline:
+    # A data container to keep a copy of an art prior to it being modified by, e.g., a character's status effects.
+    # The purpose is for comparison with base values, e.g., did AP cost go up or down overall.
+    def __init__(self, name, base_dmg, base_acc, base_stat, base_ap, base_effects):
+        self.name = name
+        self.dmg = base_dmg
+        self.acc = base_acc
+        self.stat = base_stat
+        self.ap = base_ap
+        self.effects = base_effects
+
+
+def filter_and_modify_arts(caller):
+    # Centralizes the function of sorting through the Arts table, finding those linked to a character, and then
+    # modifying them based on the character's status effect. This way, e.g., if a Berserk character's AP costs for
+    # attacks of damage less than 50 are increased by 10, this is reflected in both CmdAttack and CmdSheet.
+    # This function will be used in CmdAttack, CmdInterrupt, CmdArts, CmdListAttacks, CmdCheck, and CmdSheet.
+    arts = Arts.objects.filter(characters=caller)
+    base_arts = []
+    modified_arts = []
+    modified_normals = []
+    for art in arts:
+        # Copy.copy is used to ensure we do not modify the attack in the character's list, just this instance of it.
+        art_clean = copy.copy(art)
+        base_art = ArtBaseline(art.name, art.dmg, art.acc, art.stat, art.ap, art.effects)
+        base_arts.append(base_art)
+        # Modify the copy of the art
+        art_clean = berserk_check(caller, art_clean)
+        modified_arts.append(art_clean)
```

Comment:
Consider thinking of a different naming than `_clean` as the expression `MODIFIED_arts.append(art_CLEAN)` seems somewhat contradictory.